### ISSUE TITLE
CI Build Failures on travis

### ISSUE THREAD

**Issue by rbiswas4** [DarkEnergyScienceCollaboration/Twinkles#265]
I noticed that https://travis-ci.org/DarkEnergyScienceCollaboration/Twinkles/builds/138407187 was a failure on Master, even though the changes were only in a markdown file. 

As far as I can tell, the reason was that the initial install script of travis (with -q settings)  was running for about the entire time of the default of 20 mins provided by travis wait. So, even though nothing changed between builds, some builds were passing and others failing. I fixed this by increasing the wait time to 25 mins in https://github.com/DarkEnergyScienceCollaboration/Twinkles/commit/64a0c377323e896ef3d169ac745a04f07960d9c3, and at least in this case got a passing build (which should stay since the there is enough of a time cushion). But it only takes 20 minutes for this setup because the cache is not being downloaded:
You can see the failure on https://s3.amazonaws.com/archive.travis-ci.org/jobs/139008641/log.txt

```
$ export CASHER_DIR=$HOME/.casher
travis_time:start:0090df98
[0K$ Installing caching utilities

travis_time:end:0090df98:start=1466450961016539229,finish=1466450961127797983,duration=111258754
[0Ktravis_time:start:03ac83c1
[0K
travis_time:end:03ac83c1:start=1466450961132236978,finish=1466450961135321793,duration=3084815
[0Ktravis_time:start:04671ee1
[0K[32;1mattempting to download cache archive[0m
[32;1mfetching additionsToUpdate/cache-linux-precise-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855--compiler-gcc.tgz[0m
[32;1mfetching additionsToUpdate/cache--compiler-gcc.tgz[0m
[32;1mfetching master/cache-linux-precise-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855--compiler-gcc.tgz[0m
[32;1mfetching master/cache--compiler-gcc.tgz[0m
[31;1mcould not download cache[0m
```
 
I don't understand the working of the cache. Is this because the previous build had failed and therefore there was no cache? Can that be changed so that after a failure it is easy to test? Or does this actually mean that the only way to keep this from happening after failures is by increasing the wait time, while at other times things will be naturally speeded up?

**Comment by drphilmarshall** [comment]
Sounds like it could be. Shall we wait and see what life is like with 25
min wait times? Thanks for catching this!

**Comment by rbiswas4** [comment]
I made some more progress on this:
- Increasing this wait time, we are consistently getting success, albeit at > ~ 20 mins. The Monitor on the other hand is doing so at ~ 4 mins.
- I realized there was a problem with the cache and deleted them, if we cannot rebuild the cache obviously this is not a viable option.
- It turns out  that it was not storing the new built cache. This happens with the error message:
```
/home/travis/.casher/bin/casher:182:in `write': No space left on device - /home/travis/.casher/tar.err.log (Errno::ENOSPC)
	from /home/travis/.casher/bin/casher:182:in `tar'
	from /home/travis/.casher/bin/casher:123:in `push'
	from /home/travis/.casher/bin/casher:52:in `block in run'
	from /home/travis/.rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/timeout.rb:69:in `timeout'
	from /home/travis/.casher/bin/casher:52:in `run'
	from /home/travis/.casher/bin/casher:234:in `<main>'
```
Our cache in Twinkles seems to be 3.6 GB, while in Monitor it is 3.0 GB. I do not understand why these are different, given that they basically correspond to conda installs of the LSST software. I was also not sure what the limit allowed by travis is: maybe it is just ~ 3.0 GB and the Monitor barely makes it in, but I have not managed to get that information yet.
